File: bowtie/client/go/src/bowtiego/bowtie.go

package bowtiego

import (
	"net/http"
	"io/ioutil"
	"encoding/json"
)

const(
	SENSORS = "sensors"
	MEDIA = "media"
	NODES = "nodes"
)

type SensorData struct {
    Value interface{} `value` 
    Type string `type`
    Time string `time`
}

type NodeData map[string]SensorData

type GroupData map[string]NodeData

type BowtieServer struct {
	URL string
}

func (bs BowtieServer) GetSensor(groupId, nodeId, sensor string) (*SensorData, error) {
	resp, err := http.Get(
		bq.URL + SENSORS + "/" + 
		groupId + "/" + 
		nodeId + "/" + sensor,
	)

	if err != nil {
		return nil, err
	}

	body, readErr := ioutil.ReadAll(resp.Body)

	if readErr != nil {
		return nil, readErr
	}

	var sData SensorData

	jsonErr := json.Unmarshal(body, &sData)

	if jsonErr != nil {
		return nil, jsonErr
	}

	return sData, nil
}

File: bowtie/client/python/bowtie.py

import json
import urllib2
import time

class BowtieClient:

	def __init__(self, url="http://www.bowtie.mobi/"):

		self.url = url

		if self.url[:4] != "http":
			self.url = "http://" + self.url

		if self.url[-1] != "/":
			self.url += "/"

	def getGroup(self, groupId):
		jsonStr = urllib2.urlopen(
			self.url + "sensors/" + 
			groupId
		).read()
		return json.loads(jsonStr)

	def getNode(self, groupId, nodeId):
		jsonStr = urllib2.urlopen(
			self.url + "sensors/" + 
			groupId + "/" + 
			nodeId
		).read()
		return json.loads(jsonStr)

	def getSensor(self, groupId, nodeId, sensor):
		jsonStr = urllib2.urlopen(
			self.url + "sensors/" + 
			groupId + "/" + 
			nodeId + "/" + 
			sensor
		).read()
		return json.loads(jsonStr)
